### PR TITLE
Release jobs

### PR DESCRIPTION
**Pull request by sbesson**
Post 5.1.0 release, this PR adds the release job infrastructure to the contributing documentation for reference

### CODE REVIEW

**Comment by mtbc** [comment]
Still in progress?

File: contributing/ci-release.txt [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+        This job collects the OMERO.web integration test results
+
+        #. Receives pytest results under
+           :file:`components/tools/OmeroWeb/target/reports` from
+           :term:`OMERO-5.1-release-integration`
+        #. Generates pytest report
+
+    :jenkinsjob:`OMERO-5.1-release-training`
+
+        This jobruns the training examples
```
Comment (hflynn) [inline contributing/ci-release.txt]: Space needed between 'job' and 'runs'

File: contributing/ci-release.txt [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+
+        This job runs the robot framework tests of OMERO
+
+        #. Checks out the :envvar:`RELEASE` tag of the
+           snoopycrimecop fork of openmicroscopy.git_
+        #. Runs the robot framework tests and collect the results
+
+    :jenkinsjob:`BIOFORMATS-5.1-release`
+
+        This job triggers the release of Bio-Formats (Java & C++)
```
Comment (hflynn) [inline contributing/ci-release.txt]: & -> and

File: contributing/ci-release.txt [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+
+    :jenkinsjob:`BIOFORMATS-5.1-release-downloads`
+
+        This job builds the Bio-Formats Java downloads page
+
+        #. Checks out the `develop` branch of
+           https://github.com/openmicroscopy/ome-release.git
+        #. Runs `make clean bf`
+
+    :jenkinsjob:`BIOFORMATS-5.1-release-cpp`
```
Comment (hflynn) [inline contributing/ci-release.txt]: There seems to be a copy & paste error in these CPP build descriptions - release-cpp and release-cpp-downloads are the same except for the first point.

**Comment by mtbc** [comment]
Staged at http://www.openmicroscopy.org/site/support/contributing-staging/ci-release.html.

File: contributing/ci-release.txt [hunk 4]
```diff
@@ -0,0 +1,398 @@
...
+
+5.1.x series
+^^^^^^^^^^^^
+
+.. digraph:: OME51releasejobs
+
+  compound=true;
+  subgraph cluster0 {
+    "OME-5.1-release-push" -> "OMERO-5.1-release";
+    label = "OMERO-5.1-release-trigger";
```
Comment (mtbc) [inline contributing/ci-release.txt]: the use of the labels in this 5.1 diagram makes them not fit in the boxes

**Comment by mtbc** [comment]
I don't see any showstoppers, but I confess my eyes may have glazed over a couple of times!

**Comment by sbesson** [comment]
As an aside, regarding usages of graphs, I realized we have a Jenkins Graph Plugin. We may simply want to replace our diagrams by links to e.g. https://ci.openmicroscopy.org/view/Release/job/OME-5.1-release-trigger/lastSuccessfulBuild/BuildGraph/. Thoughts?

**Comment by mtbc** [comment]
A good idea, as I think it'd be worth it even just to save your time maintaining these by hand.

**Comment by hflynn** [comment]
Seconding the idea about linking to the Jenkins build graphs - they are more readable as well as cutting down the maintenance burden.

**Comment by sbesson** [comment]
Will do so in a follow-up PR to replace all graphs

**Comment by mtbc** [comment]
:+1:

**Comment by hflynn** [comment]
--no-rebase

**Pull request merged**